### ISSUE TITLE
Crash on Launch in 10.12.6

### ISSUE THREAD

**Issue by mangerlahn** [mangerlahn/Latest#40]
A user reported a crash occurring since at least version 0.4.3.
It has something to do with the `UpdateChecker.run()` method, according to the stack trace:

```
Date/Time:       2018-10-15 00:09:07 +0200
OS Version:      Mac OS X 10.12.6 (Build 16G1510)
Architecture:    x86_64h
Report Version:  25

Data Source:     Stackshots

Command:         Latest
Path:            /Applications/Utilities/Latest.app/Contents/MacOS/Latest
Version:         0.4.3 (9)
Parent:          launchd [1]
PID:             3146

Event:           hang
Duration:        1.10s (process was unresponsive for 267 seconds before sampling)
Steps:           11 (100ms sampling interval)

Hardware model:  iMac14,3
Active cpus:     4

Fan speed:       1395 rpm

--------------------------------------------------
Timeline format: stacks are sorted chronologically
Use -i and -heavy to re-report with count sorting
--------------------------------------------------

Heaviest stack for the main thread of the target process:
  11  start + 1 (libdyld.dylib + 21045) [0x7fffa05ad235]
  11  ??? (Latest + 9465) [0x10e2d24f9]
  11  NSApplicationMain + 780 (AppKit + 23621) [0x7fff888c1c45]
  11  -[NSStoryboard instantiateControllerWithIdentifier:] + 234 (AppKit + 8653130) [0x7fff890fc94a]
  11  -[NSNib _instantiateWithOwner:options:topLevelObjects:] + 143 (AppKit + 1167358) [0x7fff889d8ffe]
  11  -[NSNib _instantiateNibWithExternalNameTable:options:] + 696 (AppKit + 1168330) [0x7fff889d93ca]
  11  -[NSIBObjectData nibInstantiateWithOwner:options:topLevelObjects:] + 1389 (AppKit + 84102) [0x7fff888d0886]
  11  -[NSSet makeObjectsPerformSelector:] + 217 (CoreFoundation + 565241) [0x7fff8ae0dff9]
  11  -[NSWindowController _windowDidLoad] + 682 (AppKit + 748664) [0x7fff88972c78]
  11  ??? (Latest + 44471) [0x10e2dadb7]
  11  ??? (Latest + 44050) [0x10e2dac12]
  11  ??? (Latest + 161505) [0x10e2f76e1]
  11  ??? (Latest + 166841) [0x10e2f8bb9]
  11  ??? (Latest + 164187) [0x10e2f815b]
  11  ??? (Latest + 164217) [0x10e2f8179]
  11  ??? (Latest + 153908) [0x10e2f5934]
  11  ??? (Latest + 162702) [0x10e2f7b8e]
  11  -[NSLock lock] + 145 (Foundation + 32194) [0x7fff8c803dc2]
  11  __psynch_mutexwait + 10 (libsystem_kernel.dylib + 105506) [0x7fffa06dbc22]
 *11  psynch_mtxcontinue + 0 (pthread + 31211) [0xffffff7f80f079eb]
```

**Comment by beli3ver** [comment]
I get the same error. Will there be a fix for this?

**Comment by mangerlahn** [comment]
I would love to fix this, but I can't reproduce this in my side yet. Can you send me your crash report?

**Comment by beli3ver** [comment]
Ok.
So lets start. This is the output of my console app:

https://haste.tchncs.de/evoqanuhey.xml

So the strange thing is. I start the app. Then I see in the dock that it is try to start. After 5 minutes, the app is closed. But in the terminal I see the process is still running:

```
ps -ax | grep Latest                                                                                                                                                                                    ~
44901 ??         0:00.45 /Applications/Latest.app/Contents/MacOS/Latest
```

**Comment by mangerlahn** [comment]
Can you reproduce this in 0.6.1?

**Comment by mangerlahn** [comment]
I changed the update checking mechanism. This might fix this problem. Could you try this "beta" and see if it works?
https://max.codes/latest/Latest_0.6.2_Beta.zip

**Comment by mangerlahn** [comment]
This should be fixed with the now released 0.6.2. Please reopen if this is not the case :)
